src/vcpkg/commands.ci.cpp

```diff
@@ -263,6 +265,59 @@ namespace
         return has_error;
     }
 
+    struct CiSpecsResult
+    {
+        std::vector<FullPackageSpec> requested;
+        std::vector<PackageSpec> excluded;
+    };
+
+    CiSpecsResult calculate_ci_specs(const ExclusionsMap& exclusions_map,
+                                     const Triplet& target_triplet,
+                                     PortFileProvider& provider)
+    {
+        // Generate a spec for the default features for every package, except for those explicitly skipped.
+        // While `reduce_action_plan` tries to remove skipped packages as expected failures, there
+        // it is too late as we have already calculated an action plan with feature dependencies from
+        // the skipped ports.
+        CiSpecsResult result;
+        const TripletExclusions* target_triplet_exclusions = nullptr;
+        for (auto&& te : exclusions_map.triplets)
+        {
+            if (te.triplet == target_triplet)
+            {
+                target_triplet_exclusions = &te;
+                break;
+            }
+        }
+
+        for (auto scfl : provider.load_all_control_files())
+        {
+            if (!target_triplet_exclusions || !target_triplet_exclusions->exclusions.contains(scfl->to_name()))
+            {
+                result.requested.emplace_back(
+                    PackageSpec{scfl->to_name(), target_triplet},
+                    InternalFeatureSet{FeatureNameCore.to_string(), FeatureNameDefault.to_string()});
+            }
+            else
+            {
+                result.excluded.emplace_back(PackageSpec{scfl->to_name(), target_triplet});
```

Comment:
Should this also calculate exclusions for the host triplet? For example:
```
qt-complicated:x64-linux=skip
```
and
```
foo:arm64-android -> qt-complicated:$host
qt-complicated:$host -> qtbase[problem]:$host
```
where `qtbase[problem]:x64-linux` is a problem.

Reply:
No, because this calculating what we morally give to the 'vcpkg install' inside 'ci', and we don't ever add host packages there. Any handling of host dependencies needs to happen in prune_entirely_known_action_branches (formerly known as reduce_action_plan)